specs/proto/consensus.proto

```diff
@@ -4,52 +4,41 @@ import "google/protobuf/timestamp.proto";
 
 // Define wire types for consensus messages.
 
-// https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#consensusversion
-message ConsensusVersion {
-  uint64 block = 1;
-  uint64 app = 2;
-}
+// SignedMsgType is a type of signed message in the consensus.
+enum SignedMsgType {
+  SignedMsgTypeUnknown = 0;
+
+  // Votes
+  SignedMsgTypePrevote = 1;
+  SignedMsgTypePrecommit = 2;
 
-// https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#availabledataheader
-message AvailableDataHeader {
-  // array of 32-byte hashes
-  repeated bytes row_roots = 1;
-  // array of 32-byte hashes
-  repeated bytes col_roots = 2;
+  // Proposals
+  SignedMsgTypeProposal = 32;
 }
 
 message ConsensusProposal {
-  int64 height = 1;
-  int32 round = 2;
-  google.protobuf.Timestamp timestamp = 3;
-  // 32-byte hash
-  bytes last_header_hash = 4;
-  // 32-byte hash
-  bytes last_commit_hash = 5;
-  // 32-byte hash
-  bytes consensus_hash = 6;
+  SignedMsgType type = 1;
+  int64 height = 2;
+  int32 round = 3;
+  int32 pol_round = 4;
   // 32-byte hash
-  bytes state_commitment = 7;
-  uint64 available_data_original_shares_used = 8;
-  AvailableDataHeader available_data_header = 9;
+  // Proposed block header
+  bytes header_hash = 5;
```

Comment:
So if a proposal only includes the header hash, is the assumption that validators will _receive_ the header (and the DAHeader) via gossiping (if yes, why not include everything they need for validating the block in the proposal instead?), or, will they have to _request_ it (if so by whom and how).

Reply:
Tendermint and lazyledger-core currently only include the BlockID (which is the header hash + parts). Since we're removing parts, essentially they only include the header hash, not the fully header. Therefore, they must receive the block header separately _currently_.

The current specs (prior to this PR) optimize this by include all header information (including the DA header) into the proposal:

https://github.com/lazyledger/lazyledger-specs/blob/bfd7d0019a5d4f3f00dbbb263818b47be8bbb155/specs/proto/consensus.proto#L21-L37

This PR regresses that optimization to be in line with the implementation, and also prescribes regressing the optimization of including the DA header in the proposal since it's inconsistent with the exclusion of the header.

Reply:
So these fields never were redundant, right?

Reply:
They are still redundant. If a new proposal is received from an honest node, most of those values can be inferred from the current state. E.g. height must be previous height + 1. The last header hash is the previous header's hash. And so on.

Reply:
How would a DAS node perviously e.g. infer `state_commitment`, `available_data_original_shares_used`?